server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -314,19 +327,34 @@ public int markSegments(final Collection<SegmentId> segmentIds, final boolean us
   /**
    * Marks all used segments that are *fully contained by* a particular interval as unused.
    *
-   * @return the number of segments actually modified.
+   * @return Number of segments updated.
    */
   public int markSegmentsUnused(final String dataSource, final Interval interval)
+  {
+    return markSegmentsUnused(dataSource, interval, null);
+  }
+
+  /**
+   * Marks all used segments that are *fully contained by* a particular interval filtered by an optional list of versions
```

Comment:
Nit: I don't think the markdown highlighting style with asterisks would work here. You can use html tags `<b></b>` or `<i></i>` instead.